Services/Concurrency/RatedCounter.cs

```diff
@@ -177,6 +177,23 @@ public long GetPause()
             return false;
         }
 
+        public double GetThroughputForMessages()
+        {
+            if (this.timestamps.Count > 1)
+            {
```

Comment:
This code is duplicated in LogThroughput. Suggest updating LogThroughput to call this method to calculate throughput